### PR TITLE
Rspec controller borrowed_requests

### PR DESCRIPTION
**Pull request by khanhpn-2944**
WHAT
- Unit tests for controller borrowed_requests

HOW
- Fix rspec model borrowed_request
- Add rspec controller borrowed_requests

Evidence (Screenshot or Video)
Screenshot from 2022-04-20 15-46-02

### CODE REVIEW

**Comment by khanhpn-2944** [comment]
ready

File: app/views/layouts/_header.html.erb [hunk 1]
```diff
@@ -49,7 +49,7 @@
               <li>
                 <%= link_to t(".home"), root_path, class: "active" %>
               </li>
-              <% if current_user&.role %>
+              <% if current_user&.role == 1 %>
```
Comment (huongnt-2545) [inline app/views/layouts/_header.html.erb]: dùng enum để định nghĩa giá trị role trong model đi em
để 1, 2 thì biết nó là thằng nào 
doc: https://api.rubyonrails.org/v6.0/classes/ActiveRecord/Enum.html

File: spec/controllers/admin/borrowed_requests_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      before do
+        log_in admin
+        put :handle_status, params: {id: request.id, name: "accept"}
+      end
+
+      it "should flash success" do
+        expect(flash[:success]).to eq I18n.t "flash.borrowed_request.accept_success"
+      end
+
+      # it "should reload" do
```
Comment (huongnt-2545) [inline spec/controllers/admin/borrowed_requests_controller_spec.rb:57]: sao comment lại đây em?

**Comment by khanhpn-2944** [comment]
ready

File: spec/controllers/admin/borrowed_requests_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    end
+  end
+
+  describe "PUT handle_status" do
+    let!(:admin) {FactoryBot.create :user, role: 1}
+    let!(:user) {FactoryBot.create :user}
+    let!(:request) {FactoryBot.create :borrowed_request, user_id: user.id}
+
+    context "when success" do
+      before do
```
Comment (huongnt-2545) [inline spec/controllers/admin/borrowed_requests_controller_spec.rb:48]:
```suggestion
      before :all do
        log_in admin
        response {put :handle_status, params: {id: request.id, name: "accept"}}
      end
```

**Review by huongnt-2545 — APPROVED** [review]

**Pull request merged**